src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,321 @@
+
+        editTask(index, markOrUnmark);
+    }
+
+    public static void printNumberOfTask() {
+        if (tasks.size() > 1) {
+            System.out.println("You now have " + tasks.size() + " tasks.");
+        } else {
+            System.out.println("You now have " + tasks.size() + " task.");
+        }
+    }
+
+    public static void editTask(int index, String markOrUnmark) throws IOException {
+        File f = new File(filePath);
+        Scanner s = new Scanner(f);
+        String text = "";
+        int currentIndex = 0;
+        while (s.hasNext()) {
+            String currentLine = s.nextLine();
+            if (index == currentIndex) {
+                text += changeMarkValue(currentLine, markOrUnmark) + System.lineSeparator();
+            } else {
+                text += currentLine + System.lineSeparator();
+            }
+            currentIndex += 1;
+        }
+
+        FileWriter fw = new FileWriter(filePath);
+        fw.write(text);
+        fw.close();
+
+    }
+
+    public static String changeMarkValue(String currentLine, String markOrUnmark) {
+        if (markOrUnmark.equals("mark")) {
+            return currentLine.replace(" | 0 | ", " | 1 | ");
+        }
+        return currentLine.replace(" | 1 | ", " | 0 | ");
+    }
+
+    public static void addTask(String command) {
+        boolean isTodo = command.contains("todo");
+        boolean isEvent = command.contains("event");
+        boolean isDeadline = command.contains("deadline");
+
+        try {
+            String taskType = "";
+            if (isTodo) {
+                tasks.add(new Todo(command.replace("todo ", "")));
+                taskType = "todo";
+            } else if (isEvent) {
+                tasks.add(new Event(command.replace("event ", "")));
+                taskType = "event";
+            } else if (isDeadline) {
+                tasks.add(new Deadline(command.replace("deadline ", "")));
+                taskType = "deadline";
+            }
+
+            addCompleteMessage(taskType);
+            appendToFile(taskType);
+
+        } catch (EmptyDescriptionException emptyException) {
+            System.out.println("OOPS!!! The description of a todo cannot be empty.");
+        } catch (IOException e) {
+            System.out.println("Error writing to file.");
+        } catch (NoSpecficTimeException e) {
+            System.out.println("No time is entered. Please re-enter and include '/at'.");
+        } catch (NoSpecificDeadlineException e) {
+            System.out.println("No deadline is entered. Please re-enter and include '/by'. ");
+        }
+
+    }
+
+    public static void reAddTask(String command) {
+        boolean isTodo = command.contains("todo");
+        boolean isEvent = command.contains("event");
+        boolean isDeadline = command.contains("deadline");
+
+        try {
+            if (isTodo) {
+                tasks.add(new Todo(command.replace("todo", "").trim()));
```

Comment:
Try to avoid complicated expressions that might hinder readability. A good way to improve readability would be to introduce an intermediate variable(newTask or newEvent) and breaking this step down into two or more steps.

Code Quality: Avoid Complicated Expressions